### ISSUE TITLE
Cannot connect to the Jupiter server on MacOS BigSur 11.4

### ISSUE THREAD

**Issue by brunoh2015** [3-manifolds/Sage_macOS#12]
Everything was going fine and smoothly with the SageMath-9.3.rc4...
I installed the pre-release of Sagemath-9.3-v1.1 and it cannot connect anymore to the Kernel from a Jupiter Notebook

I had to go back to the previous rc4 release and everything works fine again

**Comment by NathanDunfield** [comment]
Thanks for the report.  When you installed the pre-release, did you also install the recommended setup package?

<img width="300" alt="SageMath-9_3_and_kernels_—_-bash_—_80×24" src="https://user-images.githubusercontent.com/3139290/122068644-31ea9380-cdba-11eb-86db-577512499fc6.png">

**Comment by brunoh2015** [comment]
Yes I did.

Instead of re-installing RC4 I found another way of correcting the problem: in /usr/local/share/jupyter/kernels/sagemath_9_3_all
the file kernel.json points to “var/tmp/sage-9.3-current/local/bin/sage” which does not exist.

I corrected this new pointer to “var/tmp/sage-9.3.rc4-current/local/bin/sage” which does exist
Everything works fine now, including in the Jupyter menu where I can choose two types of notebook under the name “SageMath 9.3” and “SageMath 9.3 app”

Before I manually changed the kernel.json pointer none of them worked. 
I find it strange that now the two of them work well, because in kernel.json the pointer only gives the first name …
And I do not understand why they are two such names.

**Comment by NathanDunfield** [comment]
Thanks for the details about your fix, which I expect will not entirely survive rebooting.  Specifically, if you reboot your machine and start Sage's Jupyter I predict:

1. `/var/tmp/sage-9.3-current/local/bin/sage` will exist.
2. The `SageMath 9.3` kernel will not work but the `SageMath 9.3 app` one will.  But if you re-edit the `kernel.json` file back to its original state then it will work as well.

Am I correct in my theory?

Backstory: In general, the two versions of the kernel are supposed to point to the same thing.  The reason there are two is that:

1. Users who do not install the setup package will only see `SageMath 9.3 app`.
2. Users who use a Jupyter-compatible notebook like `https://nteract.io/` or a separate Jupyter install will only see `SageMath 9.3`.

In general, the kernel ending in `app` should be the safe choice.  My theory for why this didn't work for you initially is that likely

1. You had started a Jupyter session with 9.3.rc4.  Unless you manually quit Jupyter by going to its main file-browser page and hitting the Quit button in the upper left, this will keep running in the background even after you close all the brower windows (this is harmless as it uses essentially no CPU).
2. You then installed and asked the Sage.app to start a Jupyter session with 9.3.  But it noticed that Juptyer was already running and so reconnected you to the old one.  If the old one hadn't been running it would have created `/var/tmp/sage-9.3-current` and everything would have worked without you having to edit any `kernel.json` files.

We'll have to think about the right way to deal with this issue in future upgrades.

**Comment by brunoh2015** [comment]
Thank you very much for your analysis, which happens to be almost completely correct:

1/ You are right, I did not reboot before installing the new version. After a reboot, 
> 
> /var/tmp/sage-9.3-current/local/bin/sage will exist.
does exist now. 

2/ But the directory /var/tmp/sage-9.3.rc4-current still exists after the reboot so everything works fine with the Sagemath 9.3 and the Sagemath 9.3 app kernels without changing my kernel.json pointer back to its original state.

3/ I changed as you suggested my kernel.json file to its original state. Now everything works fine as it should have been at the beginning after the installation of v1.1

4/ Thank you very much for your clear backstory.

5/ I suspect your explanation is the correct one. The only difference is that the two named kernel Sagemath 9.3 and the Sagemath 9.3 app did work after the rebooting
